internal/app/go-filecoin/internal/submodule/storage_mining_submodule.go

```diff
@@ -38,53 +48,57 @@ func NewStorageMiningSubmodule(minerAddr, workerAddr address.Address, ds datasto
 
 	smbe := piecemanager.NewStorageMinerBackEnd(storageMiner, s)
 	sbbe := postgenerator.NewSectorBuilderBackEnd(s)
+	stateViewer := state.NewViewer(b.CborStore)
 
 	modu := &StorageMiningSubmodule{
 		PieceManager:  smbe,
 		PoStGenerator: sbbe,
+		minerNode:     minerNode,
+		storageMiner:  storageMiner,
+		outbox:        m.Outbox,
```

Comment:
The submodule should not hold onto things that only its fields need. I think the outbox, chain, sectorbuilder and minerAddr fields are all only used by the poster and should be removed.